### PR TITLE
Screen templates

### PR DESCRIPTION
**Pull request by mjashanks**
1. Crete new screen > can now select from Record Detail, new Record and Record List Screen templates.
2. On Table creation, screens automaatically generated
3. Form component no longer fetches data. It must be used inside a "data"/"context" provider - i.e. a RecordDetail or a NewRecord
4. NewRecord component added - provides a record in the form `{ modelId: "...."}` - it also fetches the model
5. Bugfix: preview app not working due to appId format change 

The issues you see in the video below are due to standard-components on NPM not being the latest dev version :( ... I could have used `yarn dev`, but i wanted to show this with CSV upload

https://www.loom.com/share/6ae4596a048e48d7a1f3fa35dee17cfe

### CODE REVIEW

**Comment by mike12345567** [comment]
Hey Mike, the `app:` has changed now on master to `app_` so you might need to update your bugfix for that (also sorry for introducing the bug!)

File: packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export default {
```
Comment (shogunpurple) [inline packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js:1]: Do we need this concept? The "Empty X screen"? It seems a bit weird and doesn't really add much

File: packages/builder/src/components/userInterface/EventsEditor/actions/SaveRecord.svelte [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    const model = $backendUiStore.models.find(m => m._id === modelId)
+    parameters.modelId = modelId
+    return Object.keys(model.schema).map(k => ({
+      name: k,
+      type: model.schema[k].type,
+    }))
+  }
+
+  let schemaFields
+  $: {
```
Comment (shogunpurple) [inline packages/builder/src/components/userInterface/EventsEditor/actions/SaveRecord.svelte:77]: NAB: Most of the other svelte components in budibase follow this convention:

```
<script>
// constants

// props

// state

// reactive declarations

// functions 
</script>
```

It's a little easier to read this way, compared to jumping around.

File: packages/server/src/api/controllers/static.js.orig [hunk 3]
```diff
@@ -0,0 +1,252 @@
+const send = require("koa-send")
```
Comment (shogunpurple) [inline packages/server/src/api/controllers/static.js.orig:1]: Remove this merge file

**Review by shogunpurple — APPROVED** [review]
LGTM, just some questions for my own understanding

File: packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js (hunk 1 above)
Comment (mjashanks) [reply packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js:1]: Hmm. So, if you want to build a screen yousself, from scratch, there are the 3 "Screen Types" - the field that removed completely (aka Layout Component, aka Base Component)

These 3 screen types are
- Row Detail - loads a row into context from a `/something/:id` url
- New Row - creates a row in context, and loads the model
- Container - anything that falls outside of the above 2 etc... e.g. Lists of stuff, or screens where components do all the work (e.g. table)

"emptyContainerScreen" is the last type here - I think we still want it.

**Review by aptkingston — CHANGES_REQUESTED** [review]
Great work! This will save new builders a load of time. I've been using these changes and have just encountered a few issues which I'll detail here.

When creating a screen from one of the "empty" templates, I can't ever see any blocks I add. For example if I add a simple text component, it will be listed in the tree but never displayed in the preview. When changing the "table" setting of those screens it also does not seem to be persisted. 
Capture3

I think we should standardise the naming of the templates both in the new screen modal and in the main tree.
Capture
In the new screen modal there I think it should probably be all "row" rather than record.
In the tree I think they shoud all have a dash in the title, not just the "new" screen. So basically:
Students - New
Students - List
Students - Detail

There seem to be 2 "record detail" blocks (I think they are the new row and row details components).
Capture2

Not sure if it's a bug in master as well or not, but if you delete a screen then create a new screen, the deleted screen will come back as well. This is only visual and will disappear again if you refresh the page. Just looks like stale state somewhere.

Will be awesome to get this merged in!

**Comment by mjashanks** [comment]
Good catches there... all fixed

File: packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js (hunk 1 above)
Comment (shogunpurple) [reply packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js:1]: Would you not just create the last one from scratch though? It's not really complex enough to warrant a template IMO

**Review by aptkingston — APPROVED** [review]

File: packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js (hunk 1 above)
Comment (mjashanks) [reply packages/builder/src/builderStore/store/screenTemplates/emptyContainerScreen.js:1]: Yes - exactly! .... The "Create From Scratch" option is THIS template!

I decided just to make everything a template - even "Create from Scratch" - I will just rename to `createFromScratchTemplate.js`

**Pull request merged**